### PR TITLE
Do not reconcile images that are used by the assisted service

### PR DESCRIPTION
**Pull request by dtantsur**
We don't know how to build images suitable for it. Better ignore the
images completely and let the assisted service handle them.

Related: https://github.com/openshift/enhancements/pull/1059

### CODE REVIEW

File: cmd/controller/main.go [hunk 1]
```diff
@@ -66,11 +73,26 @@ func setupChecks(mgr ctrl.Manager) error {
...
 
 func runController(watchNamespace string, imageServer imagehandler.ImageHandler, envInputs *env.EnvInputs) error {
+	infraEnvRequirement, err := labels.NewRequirement(infraEnvLabel, selection.DoesNotExist, nil)
+	if err != nil {
+		setupLog.Error(err, "cannot create an infraenv label filter")
+		return err
+	}
+
+	cacheOptions := cache.Options{
+		SelectorsByObject: cache.SelectorsByObject{
```
Comment (dhellmann) [inline cmd/controller/main.go:83]: This appears to be saying that only resources with the label should be reconciled. I thought we wanted the opposite? Or maybe I don't understand how the cache configuration works?

Comment (zaneb) [reply cmd/controller/main.go:83]: The DoesNotExist is the key, but it cannot be inline and the naming is confusing. I'd suggest renaming `infraEnvRequirement` to `excludeInfraEnv`.

**Pull request merged**